### ISSUE TITLE
ts-jest test fails

### ISSUE THREAD

**Issue by chiragprodsmiths** [system-ui/theme-ui#1000]
TypeError: React__default.createContext is not a function

      1 | import * as React from 'react';
      2 | import { render, cleanup } from '@testing-library/react';
    > 3 | import { ThemeProvider } from 'theme-ui';
        | ^
      4 | import Login from '../index';
      5 | import { theme } from '../../../theme/index';
      6 | 

      at Object.<anonymous> (node_modules/@mdx-js/react/dist/cjs.js:117:33)
      at Object.<anonymous> (node_modules/@theme-ui/mdx/dist/index.js:8:15)
      at Object.<anonymous> (node_modules/theme-ui/dist/index.js:3:11)
      at Object.<anonymous> (src/components/login/tests/login.form.spec.tsx:3:1)

**Comment by chiragprodsmiths** [comment]
@jxnblk @mxstbr 

Hey Guys can anyone help me out. this has stopped me write any tests. 

using ts-jest, jest, typescript.

**Comment by hasparus** [comment]
Hey @chiragprodsmiths, could you share your code with me?

**Comment by chiragprodsmiths** [comment]
> Hey @chiragprodsmiths, could you share your code with me? Do you have a reproduction repo, or happen to be working on open source? I'll try to help you out.
> 
> Which version of `theme-ui` are you using?

https://github.com/chiragprodsmiths/theme-ui-ts-jest

checkout readme for spec file that has issue
https://github.com/chiragprodsmiths/theme-ui-ts-jest/blob/master/README.md

the issue only happens when i bring in theme-ui.

**Comment by hasparus** [comment]
Okay, I took a short look at your repository.

What's interesting, I got a different error in your repo. I'm puzzled.
`ThemeProvider` is supposedly undefined for me.

![image](https://user-images.githubusercontent.com/15332326/84644184-1d519200-aeff-11ea-8feb-cc7f4220da6e.png)


The problem appears on `theme-ui` version **0.3.1**, and upgrading to **0.4.0-alpha.3** doesn't fix it.

In you case, I'd consider changing `import React from 'react'` in _node_modules/@theme-ui/core_ to `import * as React from 'react'`.

**Comment by chiragprodsmiths** [comment]
> Okay, I took a short look at your repository.
> 
> What's interesting, I got a different error in your repo. I'm puzzled.
> `ThemeProvider` is supposedly undefined for me.
> 
> ![image](https://user-images.githubusercontent.com/15332326/84644184-1d519200-aeff-11ea-8feb-cc7f4220da6e.png)
> 
> The problem appears on `theme-ui` version **0.3.1**, and upgrading to **0.4.0-alpha.3** doesn't fix it.
> 
> In you case, I'd consider changing `import React from 'react'` in _node_modules/@theme-ui/core_ to `import * as React from 'react'`.

@hasparus  its undefined as i have declared as I have this
**moduleNameMapper: {
    'theme-ui': '<rootDir>/node_modules/theme-ui/dist/index.js',
  },**
here => https://github.com/chiragprodsmiths/theme-ui-ts-jest/blob/master/jest.config.js

if I don't modulename it then it gives me 
**TypeError: React__default.createContext is not a function**

and yes touching node_modules is of course we don't want

**Comment by chiragprodsmiths** [comment]
@hasparus 

any feedback on this please. I am unable to unit testing in my app & I am really worried about it

**Comment by chiragprodsmiths** [comment]
@hasparus  also I changed `import React from 'react'` to `import *  as React from 'react'` in node_modules/@theme-ui/core
but still same error

**Comment by chiragprodsmiths** [comment]
![image](https://user-images.githubusercontent.com/49074609/87020224-2d544d00-c1f1-11ea-88f9-1bdefaf1853d.png)

the problem is with mdx package and its dependency on @mdx/react.

**Comment by hasparus** [comment]
Okay, I've dived into your node_modules.

The problem appears in line

```
var MDXContext = React__default.createContext({});
```
at (node_modules/@mdx-js/react/dist/cjs.js:117:33)

`React__default` is created by `interopDefault(React)`

where `interopDefault` is

```
function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }
```

React doesn't have a default export, so `createContext` should be there.

However, when I console.logged React in this file, I got an empty object.

Is Jest misconfigured? It turns out it is.

🌟 **TLDR**: I removed `.` from `moduleDirectories` array in your Jest config. React is properly imported. The problem dissapears.

```
  moduleDirectories: ['node_modules', 'src'],
```

There is another problem left

```
    Cannot find module 'theme/theme.fontface' from 'src/theme/theme.typography.ts'
```

It can be solved by changin `moduleNameMapper` Jest option to match your TSConfig `paths`.
